### PR TITLE
Debug info for unreachable constants and variables

### PR DESCRIPTION
**Pull request by tmat**
**Customer scenario**

The main motivation of this change is to simplify validation of conversion between Portable PDBs and Windows PDBs. 
Currently, local variable debug info emitted to Portable PDB differs from Windows PDB -- the order is different and variables defined in an unreachable code are emitted in one and not the other. An inconsistency also exists between debug info emitted for local variables and for local constants. This change makes them all consistent and thus easy to compare. 

**Bugs this fixes:**

N/A

**Workarounds, if any**

None.

**Risk**

Small.

**Performance impact**

None.

**Is this a regression from a previous update?**

**Root cause analysis:**

**How was the bug found?**

Testing PDB converter.

### CODE REVIEW

**Comment by tmat** [comment]
@dotnet/roslyn-compiler Please review.

File: src/Compilers/CSharp/Test/Emit/PDB/PDBTupleTests.cs [hunk 1]
```diff
@@ -262,5 +263,83 @@ static void F(System.Collections.Generic.IEnumerable<(int a, int b)> ie)
...
+}
+";
+            var c = CreateStandardCompilation(source, new[] { ValueTupleRef, SystemRuntimeFacadeRef }, options: TestOptions.DebugDll);
+            var v = CompileAndVerify(c);
+            v.VerifyIL("C.F", @"
+{
+  // Code size        5 (0x5)
+  .maxstack  1
+  .locals init ((int a, int b)[] V_0, //v1
+                (int a, int b)[] V_1, //v2
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/PDB/PDBTupleTests.cs]: I'm surprised that we emit the second and third locals at all. Should that be tracked by an issue?

File: src/Compilers/CSharp/Test/Emit/PDB/PDBTupleTests.cs [hunk 2]
```diff
@@ -262,5 +263,83 @@ static void F(System.Collections.Generic.IEnumerable<(int a, int b)> ie)
...
+        </encLocalSlotMap>
+      </customDebugInfo>
+      <sequencePoints>
+        <entry offset=""0x0"" startLine=""5"" startColumn=""5"" endLine=""5"" endColumn=""6"" />
+        <entry offset=""0x1"" startLine=""6"" startColumn=""9"" endLine=""6"" endColumn=""36"" />
+        <entry offset=""0x3"" startLine=""9"" startColumn=""9"" endLine=""9"" endColumn=""20"" />
+      </sequencePoints>
+      <scope startOffset=""0x0"" endOffset=""0x5"">
+        <local name=""v1"" il_index=""0"" il_start=""0x0"" il_end=""0x5"" attributes=""0"" />
+        <local name=""v2"" il_index=""1"" il_start=""0x0"" il_end=""0x5"" attributes=""0"" />
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/PDB/PDBTupleTests.cs]: Could you clarify why v2 is described, but not v3 (despite v3 being in the IL too)?

**Comment by jcouv** [comment]
> This change makes them all consistent and thus easy to compare.

Do you want to include a test that emits both kinds of PDBs and compares them? They should be identical, right?

**Comment by tmat** [comment]
@jcouv I'm working on the tests. Future PR...

File: src/Compilers/CSharp/Test/Emit/PDB/PDBTupleTests.cs (hunk 1 above)
Comment (tmat) [inline src/Compilers/CSharp/Test/Emit/PDB/PDBTupleTests.cs]: Yes, me too. @VSadov said we don't bother to remove variables defined in unreachable code.

File: src/Compilers/CSharp/Test/Emit/PDB/PDBTupleTests.cs (hunk 2 above)
Comment (tmat) [inline src/Compilers/CSharp/Test/Emit/PDB/PDBTupleTests.cs]: v3 is defined in unreachable code.

File: src/Compilers/CSharp/Test/Emit/PDB/PDBTupleTests.cs (hunk 1 above)
Comment (tmat) [inline src/Compilers/CSharp/Test/Emit/PDB/PDBTupleTests.cs]: Issue https://github.com/dotnet/roslyn/issues/17947 track this inconsistency in the compiler.

**Comment by tmat** [comment]
test windows_release_vs-integration_prtest please

**Comment by tmat** [comment]
@MattGertz Oops, was too eager to merge. Forgot ask most approval :-|

**Comment by tmat** [comment]
@MattGertz Thanks!

**Pull request merged**